### PR TITLE
[ISSUE-2457] Add support for CTAS(Create Table As Select) statement

### LINKED ISSUE #2457: Support create table as/like query
**Summary**

```
mysql> create table test_b like test_a ;
ERROR 1105 (HY000): Code: 1002, displayText = sql parser error: Expected end of statement, found: like.
mysql>  create table (number UInt64) as select number from numbers(100);
ERROR 1105 (HY000): Code: 1002, displayText = sql parser error: Expected identifier, found: (.
```

Refer: https://dev.mysql.com/doc/refman/8.0/en/create-table-like.html

### PR DESCRIPTION
**Pull request by junli1026**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Add support for create table as select statement

## Changelog

- Improvement

## Related Issues

Fixes #2457

## Test Plan

Unit Tests

Stateless Tests

## DOC 
Screen Shot 2021-12-12 at 4 59 18 PM
Screen Shot 2021-12-12 at 5 00 38 PM

### CODE REVIEW

**Comment by junli1026** [comment]
Thanks @dantengsky for adding the 'query' field in the sql parser. It is very helpful!

**Comment by BohuTANG** [comment]
@mergifyio update

**Comment by dantengsky** [comment]
look great to me, thanks @junli1026 !

@BohuTANG 

do we need to clarify(in doc) that the "append columns" semantic of MySQL CTAS are not supported(by design) in our impl?

from https://dev.mysql.com/doc/refman/8.0/en/create-table-select.html, 
> Notice that the columns from the SELECT statement are appended to the right side of the table, not overlapped onto it. Take the following example
```



mysql> SELECT * FROM foo;
+---+
| n |
+---+
| 1 |
+---+

mysql> CREATE TABLE bar (m INT) SELECT n FROM foo;
Query OK, 1 row affected (0.02 sec)
Records: 1  Duplicates: 0  Warnings: 0

mysql> SELECT * FROM bar;
+------+---+
| m    | n |
+------+---+
| NULL | 1 |
+------+---+
1 row in set (0.00 sec)
```
personally, I never used this feature of MySQL. not mention this in the doc is ok for me.

**Comment by dantengsky** [comment]
> Thanks @dantengsky for adding the 'query' field in the sql parser. It is very helpful!

Yes, it is! but it is contributed by @AngleNet in PR #3356, thanks @AngleNet !

File: tests/suites/0_stateless/05_0000_ddl_create_tables.sql [hunk 1]
```diff
@@ -16,11 +16,18 @@ create table t2(a int,b int) engine=NotExists; -- {ErrorCode 4003}
...
 CREATE DATABASE db1;
 CREATE DATABASE db2;
 CREATE TABLE db1.test1(a INT, b INT);
+INSERT INTO db1.test1 VALUES (1, 1), (2, 2), (3, 3);
+-- test 'create table like' statement, expect db2.test2 has the same schema with db1.test1
 CREATE TABLE db2.test2 LIKE db1.test1;
 INSERT INTO db2.test2 VALUES (3, 5);
 SELECT a+b FROM db2.test2;
+-- test 'create table as select' statement, expect db2.test3 has the data from db1.test1 with casting
+CREATE TABLE db2.test3(x Varchar, y Varchar) AS SELECT * FROM db1.test1;
```
Comment (BohuTANG) [inline tests/suites/0_stateless/05_0000_ddl_create_tables.sql:29]: Looks great :+1: 
Do you have the plan change this test to 
```
CREATE TABLE db1.test1(a INT, b INT) ENGINE=MEMORY;

... ...

CREATE TABLE db2.test3(x Varchar, y Varchar) ENGINE=FUSE AS SELECT * FROM db1.test1;
... ...
```
?

**Comment by BohuTANG** [comment]
@dantengsky 
This will be addressed in #3439

**Comment by junli1026** [comment]
> > Thanks @dantengsky for adding the 'query' field in the sql parser. It is very helpful!
> 
> Yes, it is! but it is contributed by @AngleNet in PR #3356, thanks @AngleNet !

Thank you @dantengsky all the same for pointing me to @AngleNet 

Thank you @AngleNet  for you contribution !

**Comment by BohuTANG** [comment]
/LGTM

Thank you @junli1026 !

**Pull request merged**